src/services/LibraryService.ts

```diff
@@ -22,9 +21,9 @@ export class LibraryService implements Service<fhir4.Library> {
     logger.info(`GET /Library`);
     const { query } = req;
     logger.debug(`Request Query: ${JSON.stringify(query, null, 2)}`);
-    validateSearchParams(query);
-    const parsedQuery = getMongoQueryFromRequest(query);
-    const entries = await findResourcesWithQuery<fhir4.Library>(parsedQuery, 'Library');
+    const parsedQuery = parseRequestSchema<typeof LibrarySearchArgs>(query, LibrarySearchArgs);
```

Comment:
Isn't this generic inferred by TypeScript given that the generic is the type of the second parameter? You should be able to remove this

```suggestion
    const parsedQuery = parseRequestSchema(query, LibrarySearchArgs);
```

and it should still work. That's the beauty of generics!